kratos/geometries/nurbs_shape_function_utilities/nurbs_utilities.h

```diff
@@ -66,8 +66,17 @@ namespace NurbsUtilities
         const Vector& rKnots,
         const double ParameterT)
     {
+        // Check if the ParameterT (gauss point) is coincident to a knot (laying on an edge of a knot span)
+        double parameter_t_corrected = ParameterT;
+        for (unsigned i = 0; i<rKnots.size(); i++) {
+            if (std::abs(ParameterT-rKnots[i]) < 1e-13) {
```

Comment:
> `1e-13`

magic tolerances tend to break at the most unexpected times, and debugging hard-coded ones is a real pain in the ass. I suggest passing them in from the outside, or defining them in some central location that you'll find easier later.